source/docs/software/basic-programming/git-getting-started.rst

```diff
@@ -0,0 +1,250 @@
+
+Repository
+----------
+
+A Git repository is a data structure containing the structure, history, and files of a project.
+
+Git repositories consist of:
+
+- A ``.git`` folder. This folder contains the various information about the repository. 
+- A ``.gitignore`` file. This file contains the files or directories that you do *not* want included when you commit.
+- Files and folders. This is the main content of the repository.
+
+Creating the repository
+^^^^^^^^^^^^^^^^^^^^^^^
+
+You can store the repository locally, or through the cloud. `Github <https://github.com/>`_ is a popular free hosting service. Numerous developers use it, and that's what this tutorial will use.
+
+Creating a Github Account
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Go ahead and create a Github account by visiting the `website <https://github.com>`_ and following the own screen prompts.
+
+.. image:: images/image1.png
+
+Local Creation
+~~~~~~~~~~~~~~
+
+After creating and verifying your account, you'll want to visit the homepage. It'll look similar to the shown image.
+
+.. image:: images/image2.png
+
+Click the plus icon in the top right.
+
+.. image:: images/image3.png
+
+Then click *"New Repository"*
+
+.. image:: images/image4.png
+
+Fill out the appropriate information, and then click *"Create repository"*
+
+.. image:: images/image5.png
+
+You should see a screen similar to this
+
+.. image:: images/image6.png
+
+Now you'll want to open a powershell window and navigate to your project directory. An excellent tutorial can be found `here <https://programminghistorian.org/en/lessons/intro-to-powershell>`__
+
+.. image:: images/image7.png
+
+In the below example, we created a file called ``README.md`` with the contents of ``# Example Repo``. More details on the various commands can be found in the subsequent sections.
+
+.. image:: images/image8.png
+
+Commits
+-------
+
+Github repository are primarily composed of commits. Commits are saved states or *versions* of code. 
+
+In the previous example, we created a file called README.md. Open that file in your favorite text editor and edit a few lines. After tinkering with the file for a bit, simply save and close. Navigate to powershell and type the following commands.
+
+1. ``git add README.md``
+2. ``git commit -m "example commit"``
+3. ``git push``
+
+.. image:: images/image9.png
+
+Git Pull
+^^^^^^^^
+
+.. note:: ``git fetch`` can be used when the user does not wish to automatically merge into the current working branch
+
+This command retrieves the history or commits from the remote repository. When the remote contains work you do not have, it will attempt to automatically merge. See :ref:`docs/software/basic-programming/git-getting-started:Merging`.
+
+Run: ``git pull``
+
+Git Add
+^^^^^^^
+
+This command adds a selected file(s) to a commit. To commit every file/folder that isn't excluded via *gitignore*.
+
+Run: ``git add .``.
```

Comment:
This only adds every file if you’re in the root directory. It’s also a terrible practice because people will accidentally add files they don’t mean to